crates/pathfinder/src/sequencer.rs

```diff
@@ -411,13 +434,63 @@ impl ClientApi for Client {
         retry(|| async {
             let resp = self
                 .inner
-                .get(self.build_query("get_contract_addresses", &[]))
+                .get(self.build_query(&["feeder_gateway", "get_contract_addresses"], &[]))
                 .send()
                 .await?;
             parse(resp).await
         })
         .await
     }
+
+    /// Adds a transaction invoking a contract.
+    #[tracing::instrument(skip(self))]
+    async fn add_invoke_transaction(
+        &self,
+        call: Call,
+        max_fee: Fee,
+        version: TransactionVersion,
+    ) -> Result<reply::add_transaction::InvokeResponse, SequencerError> {
+        let req = request::add_transaction::AddTransaction::Invoke(
+            request::add_transaction::InvokeFunction {
+                contract_address: call.contract_address,
+                entry_point_selector: call.entry_point_selector,
+                calldata: call.calldata,
+                max_fee,
+                version,
+                signature: call.signature,
+            },
+        );
+        let resp = self
+            .inner
+            .post(self.build_query(&["gateway", "add_transaction"], &[]))
+            .json(&req)
+            .send()
+            .await?;
+        parse(resp).await
+    }
+
+    /// Deploys a contract.
+    #[tracing::instrument(skip(self, contract_definition))]
+    async fn add_deploy_transaction(
+        &self,
+        contract_address_salt: ContractAddressSalt,
+        constructor_calldata: Vec<ConstructorParam>,
+        contract_definition: ContractDefinition,
+    ) -> Result<reply::add_transaction::DeployResponse, SequencerError> {
+        let req =
+            request::add_transaction::AddTransaction::Deploy(request::add_transaction::Deploy {
+                contract_address_salt,
+                contract_definition,
+                constructor_calldata,
+            });
+        let resp = self
+            .inner
+            .post(self.build_query(&["gateway", "add_transaction"], &[]))
+            .json(&req)
+            .send()
+            .await?;
+        parse(resp).await
```

Comment:
Thinking about retries... Maybe the declare is safe to retry, or do we not want to do that?

Unsure if invoke is safe to retry.

Reply:
I don't think it makes sense for us to retry. That's just masking sequencer errors from our client I guess. If retries are needed those should be implemented in the client I think.

Reply:
Could comment on why not in the methods, so we remember it when this comes up :)

Reply:
Absolutely! I'll add comments.

Reply:
Done.